xla/service/gpu/gpu_fusible.cc

```diff
@@ -885,12 +885,6 @@ static void GetFusionRootsRec(const HloInstruction* root,
       GetFusionRootsRec(root->operand(i), out);
     }
   } else {
-    if (!out.empty() && out.back() == root) {
-      return;
-    }
-    CHECK(!absl::c_linear_search(out, root))
```

Comment:
I think the check still makes sense, but in line 881 we assume that if we have a GetTupleElement op, it has as operand a Tuple op (which is not true in case of DynamicAddressFusion). So I think a better fix might be to change line 881 to ```if (root->opcode() == HloOpcode::kGetTupleElement && root->operand(0)->opcode() == HloOpcode::kTuple)```

See also https://github.com/openxla/xla/commit/2204f157da667491c1baa4f2f37a883cfc73068e for a similar recent fix.

Reply:
Does this look okay? I removed the check from line 888 because it makes sense to not allow returning same values, but why we should be allowing them in adjacent elements in tuple is not clear to me.

Reply:
Yes, looks ok to me. I also don't understand why we try to allow a duplicate root in case it is encountered twice in a row in the recursive search. Seems arbitrary to me, and we should also check fail in that case.